src/farcasterd/runtime.rs

```diff
@@ -857,59 +857,56 @@ impl Runtime {
+                        }
+                        let msg = format!(
+                            "{} {}",
+                            "Public offer registered, please share with taker: ".bright_blue_bold(),
+                            serialized_offer.bright_yellow_bold()
+                        );
+                        info!(
+                            "{}: {:#}",
+                            "Public offer registered".bright_green_bold(),
+                            pub_offer_id.bright_yellow_bold()
+                        );
+                        report_to.push((
+                            Some(source.clone()),
+                            Request::Success(OptionDetails(Some(msg))),
+                        ));
+                        self.arb_addrs.insert(pub_offer_id, arbitrating_addr);
+                        self.acc_addrs
+                            .insert(pub_offer_id, monero::Address::from_str(&accordant_addr)?);
+                        Ok(())
+                    }
                     Err(err) => {
                         error!("{}", err.err());
-                        return Err(err);
+                        Err(err)
                     }
-                }
-                report_to.push((
-                    Some(source.clone()),
-                    resp.into_progress_or_failure(),
-                    // Request::Progress(format!(
-                    //     "Node {} listens for connections on {}",
-                    //     self.node_id, remote_addr
-                    // )),
-                ));
-                let node_id = self.node_ids.get(&offer.id()).cloned().unwrap();
-                let public_offer = offer.to_public_v1(node_id, public_addr.into());
-                let pub_offer_id = public_offer.id();
-                let serialized_offer = public_offer.to_string();
-                if self.public_offers.insert(public_offer) {
-                    let msg = format!(
-                        "{} {}",
-                        "Public offer registered, please share with taker: ".bright_blue_bold(),
-                        serialized_offer.bright_yellow_bold()
-                    );
-                    info!(
-                        "{}: {:#}",
-                        "Public offer registered".bright_green_bold(),
-                        pub_offer_id.bright_yellow_bold()
-                    );
-                    report_to.push((
-                        Some(source.clone()),
-                        Request::Success(OptionDetails(Some(msg))),
-                    ));
-                    self.arb_addrs.insert(pub_offer_id, arbitrating_addr);
-                    self.acc_addrs
-                        .insert(pub_offer_id, monero::Address::from_str(&accordant_addr)?);
-                } else {
-                    let msg = "This Public offer was previously registered";
-                    warn!("{}", msg.err());
+                };
+                if res.is_err() {
```

Comment:
use `if let Err(err) = res {` instead and avoid `.err().unwrap()` later.

Reply:
Done in 241291397afca2d3b87c1c5abfbd86b4273bb000